lib/faraday.rb

```diff
@@ -78,8 +79,8 @@ class << self
     #   Faraday.new url: 'http://faraday.com',
     #               params: { page: 1 }
     #   # => Faraday::Connection to http://faraday.com?page=1
-    def new(url = nil, options = nil, &block)
-      options = options ? default_connection_options.merge(options) : default_connection_options
+    def new(url = nil, options = {}, &block)
+      options = default_connection_options.merge(options)
```

Comment:
FWIW this will raise an exception if someone passes nil to `options`. I don't know why anyone would do that though. It also needlessly creates a new hash object for an optional argument. Probably not a big deal though.

Reply:
It's a good, I can fix the first point by defaulting to `nil` and doing `default_connection_options.merge(options || {})`, but an empty hash will still be initialised every time.